packages/flutter/lib/src/rendering/semantics.dart

```diff
@@ -0,0 +1,456 @@
+          result.flags.canBeTapped = result.flags.canBeTapped || node.canBeTapped;
+          result.flags.canBeLongPressed = result.flags.canBeLongPressed || node.canBeLongPressed;
+          result.flags.canBeScrolledHorizontally = result.flags.canBeScrolledHorizontally || node.canBeScrolledHorizontally;
+          result.flags.canBeScrolledVertically = result.flags.canBeScrolledVertically || node.canBeScrolledVertically;
+          result.flags.hasCheckedState = result.flags.canBeLongPressed || node.hasCheckedState;
+          result.flags.isChecked = result.flags.isChecked || node.isChecked;
+          if (node.label != '')
+            result.strings.label = result.strings.label.isNotEmpty ? '${result.strings.label}\n${node.label}' : node.label;
+          return true; // continue walk
+        });
+        // and we pretend to have no children
+      } else {
+        if (_children != null) {
+          for (SemanticsNode child in _children)
+            children.add(child._serialize());
+        }
+      }
+      result.children = children;
+      _dirty = false;
+    }
+    return result;
+  }
+
+  static void sendSemanticsTreeTo(engine.SemanticsClient client) {
+    for (SemanticsNode oldNode in _detachedNodes) {
+      // The other side will have forgotten this node if we even send
+      // it again, so make sure to mark it dirty so that it'll get
+      // sent if it is resurrected.
+      oldNode._dirty = true;
+    }
+    _detachedNodes.clear();
+    if (_dirtyNodes.isEmpty)
+      return;
+    _dirtyNodes.sort((SemanticsNode a, SemanticsNode b) => a.depth - b.depth);
+    for (int index = 0; index < _dirtyNodes.length; index += 1) {
+      // we mutate the list as we walk it here, which is why we use an index instead of an iterator
+      SemanticsNode node = _dirtyNodes[index];
+      assert(node.parent == null || !node.parent.mergeAllDescendantsIntoThisNode || node.mergeAllDescendantsIntoThisNode);
+      if (node.mergeAllDescendantsIntoThisNode) {
+        // if we're merged into our parent, make sure our parent is added to the list
+        if (node.parent != null && node.parent.mergeAllDescendantsIntoThisNode)
+          node.parent._markDirty(); // this can add the node to the dirty list
+        // make sure all the descendants are also marked, so that if one gets marked dirty later we know to walk up then too
+        if (node._children != null)
+          for (SemanticsNode child in node._children)
+            child.mergeAllDescendantsIntoThisNode = true; // this can add the node to the dirty list
+      }
+      assert(_dirtyNodes[index] == node); // make sure nothing went in front of us in the list
+    }    
+    _dirtyNodes.sort((SemanticsNode a, SemanticsNode b) => a.depth - b.depth);
+    List<engine.SemanticsNode> updatedNodes = <engine.SemanticsNode>[];
+    for (SemanticsNode node in _dirtyNodes) {
+      assert(node.parent?._dirty != true);
+      if (node._dirty)
```

Comment:
I'll add a comment to explain it in the code, but for now: `_serialize()` does a deep serialisation of all contiguous dirty nodes. So it can end up cleaning more than just the top node of the serialisation.